### ISSUE TITLE
Conrastive Loss

### ISSUE THREAD

**Issue by old-school-kid** [keras-team/keras-cv#1851]
**Short Description**
Pair loss is used in contrastive pre-training (backbone of CLIP) and siamese network which is kind of very important in the field of computer vision and multi modal learning. We dont have to create pairwise data for contrastive loss, rather can take samples from minibatches. Also if we want to add this feature should we directly add the contrastive loss (hardcode to only single label) or use more of a n-pair loss where the logit is calculated by the user and plugged into the loss function?

**Papers**
[CLIP](https://arxiv.org/abs/2103.00020) Citations: 5558
[Siamese Network](https://www.cs.cmu.edu/~rsalakhu/papers/oneshot1.pdf) Citations: 4135

**Existing Implementations**
[Tensorflow addon implementation](https://github.com/tensorflow/addons/blob/v0.20.0/tensorflow_addons/losses/contrastive.py)

**Other Information**
Do I want to contribute a PR?: yes

**Comment by old-school-kid** [comment]
Closing this as got it transferred to [here](https://github.com/keras-team/keras/issues/18203#issue-1745857040)
